Paper: Qiita: report of progress towards an open access microbiome data analysis and visualization platform (Qiita)

adamrp opened the pull request:
Advances in sequencing, proteomics, transcriptomics and metabolomics are giving us new insights into the microbial world and dramatically improving our ability to understand microbial community composition and function at high resolution. These new technologies are generating vast amounts of data, even from a single study or sample, leading to challenges in storage, representation, analysis, and integration of the disparate data types.

[Qiita](https://github.com/biocore/qiita) aims to be the leading platform to store, analyze, and share multi-omics data. Qiita is BSD-licensed, unit-tested, and adherent to PEP8 style guidelines. New code additions are reviewed by multiple developers and tested using Travis CI. This approach opens development to the largest possible number of experts in "-omics" fields. The heterogeneous data generated by these disciplines led us to use a combination of Redis, PostgreSQL, BIOM (McDonald et al. 2012), and HDF5 for relational and hierarchical storage. The compute backend is provided by [IPython](http://ipython.org/)’s parallel framework. In addition, the project depends on mature Python packages such as [Tornado](http://www.tornadoweb.org/en/stable), [click](http://click.pocoo.org/), [scipy](http://www.scipy.org/), [numpy](http://www.numpy.org/), and [scikit-bio](http://scikit-bio.org/) among others. Most notably, the analysis pipeline is provided by [QIIME](http://qiime.org/), with [EMPeror](https://github.com/biocore/emperor) serving as the visualization platform for high-dimensional ordination plots, which can be recolored interactively and manipulated using the sample metadata.  

By providing the database and compute resources at [qiita.microbio.me](http://qiita.microbio.me/) to the global community of microbiome researchers, Qiita alleviates the technical burdens, such as familiarity with the command line or access to compute power, that are typically limiting for researchers studying microbial ecology, while at the same time promoting an open access culture. Because Qiita is entirely open source and highly scalable, developers can inspect, customize, and extend it to suit their needs regardless of whether it is deployed as a desktop application or as a shared resource.

[Edit by jaberg] http://nipy.bic.berkeley.edu:5000/download/14

adina commented on papers/qiita_development_team/qiita_development_team.rst:
```diff
@@ -0,0 +1,562 @@
+.. class:: keywords
+
+   Microbiome, multi-omics, open science.
+
```
I would suggest that instead of multi-omics, you include metagenomics, metatranscriptomics, and metaproteomics (if applicable) as these will be the search terms used (rarely multi-omics)

adamrp replied:
We have supplemented multi-omics with the terms you suggested (and added metabolomics, as well). Thanks!

adina commented on papers/qiita_development_team/qiita_development_team.rst:
```diff
@@ -0,0 +1,562 @@
+and broader analyses have begun to characterize the potent effects that
+microbial communities can impart on both host-associated ([Atr07]_,
+[Atr03]_) and environmental ([Atr09]_) health. The myriad
+techniques that can be used to characterize each individual sample allow
```
I would disagree that sequencing technologies have led to broader analyses.  In many ways it has led to very focused analyses on "biological sequencing data".  I would suggest some re-word here.  Suggestion of something along the lines of:  The costs of sequencing tech has dramatically changed the type, volume, velocity of the data that can now be integrated with other datsets...

adamrp replied:
We have reworded to this to address your concern.

adina commented on papers/qiita_development_team/qiita_development_team.rst:
```diff
@@ -0,0 +1,562 @@
+
+The genome of an organism is the set of genes that the organism possesses; the
+“metagenome” of a group of organisms is the union of the genomes in the group.
+Since the genome of an organism defines the organism’s biological capabilities,
```
I dont think its correct to call a metagenome the union of genomes - rather it describes that a metagenome is DNA (genes or otherwise) directly extracted from an environmental sample.   A genome is the DNA originating from a single organism (it also contains more than genes).

adamrp replied:
We changed the wording to clarify this subtle distinction.

adina commented on papers/qiita_development_team/qiita_development_team.rst:
```diff
@@ -0,0 +1,562 @@
+the organisms in a sample capable of doing, collectively? When a metagenomic
+analysis is performed, the metagenome is fragmented into small pieces,
+sequenced in a massively parallel fashion, and genes are identified by
+comparison to references containing known genes. This technique results in a
```
I struggle with the technical precision of the description vs. what is tractable to broad understanding.  I think it is important to be technically correct though still.  The sequencing technology is what drives the need to "fragment" environmental DNA.  Not all sequencing technologies fragment into small pieces and sequence in massively parallel.  The metagenomic analysis is not driving the "fragmentation however."

adamrp replied:
Reworded to clarify that current techniques fragment the metagenome into small pieces, but that this is not *necessary* for metagenomic analysis. To the best of our knowledge, this is the most common technique (which we aim to support in Qiita); are you familiar with other techniques that do not do this?

adina commented on papers/qiita_development_team/qiita_development_team.rst:
```diff
@@ -0,0 +1,562 @@
+highly detailed view, but is relatively expensive due to the amount of
+sequencing that must be performed and the computational effort required
+([Atr13]_). A less detailed (but much cheaper and still very useful)
+characterization of a microbial community can be attained merely by performing
```
Some comment on why sequencing volume and computational effort is high needs to be determined.

adina commented on papers/qiita_development_team/qiita_development_team.rst:
```diff
@@ -0,0 +1,562 @@
+targeted sequencing of marker genes. Sequences from marker genes are commonly
+grouped by similarity into operational taxonomic units (OTUs), groupings that
+might correspond to species, or genera, or classes, etc. A powerful way to
+identify the OTUs present in a sample is to amplify and sequence components of
```
"merely" is not a term here that like.  This is not a trivial thing, its actually quite complicated and amazing and I think this takes away from that.  There are also disadvantages to this technique that should be considered and stated.

adamrp replied:
We have removed the word "merely", but we feel that delving into the relative pros and cons of target gene sequencing is outside the scope of this introduction.

adina commented on papers/qiita_development_team/qiita_development_team.rst:
```diff
@@ -0,0 +1,562 @@
+grouped by similarity into operational taxonomic units (OTUs), groupings that
+might correspond to species, or genera, or classes, etc. A powerful way to
+identify the OTUs present in a sample is to amplify and sequence components of
+the ribosomal operon in a sample (rather than all of the genes). The ribosome
```
Suggest not using the word operon but just use genes to try to keep jargon at minimal

adamrp replied:
Reworded to remove the mention of "operon"

adina commented on papers/qiita_development_team/qiita_development_team.rst:
```diff
@@ -0,0 +1,562 @@
+evolution, mutations accrue at a relatively consistent rate, allowing it to be
+thought of as a “molecular clock” that provides phylogenetic information about
+the organism it came from ([Atr15]_). This technique is called amplicon
+sequencing. In bacteria and archaea, the 16S small subunit ribosomal gene is
```
16S rRNA gene (for bacteria at least) amplicon sequencing.  Amplicon sequencing can describe any targeted gene.

adamrp replied:
This oversimplification has been corrected.

adina commented on papers/qiita_development_team/qiita_development_team.rst:
```diff
@@ -0,0 +1,562 @@
+metagenomics help answer the questions of who's there, and what are they
+capable of doing, transcriptomics help answer the question what are they
+actually doing right now? Genes that are “on” can be recognized by the presence
+of mRNA transcripts that identifiably correspond to the gene.  Sequencing these
```
Gene expression should be the focus rather than what they are doing right now.  For example, an expressed gene does not equal function.

adamrp replied:
Thanks for catching that, we have corrected it.

adina commented on papers/qiita_development_team/qiita_development_team.rst:
```diff
@@ -0,0 +1,562 @@
+environments? Similar to proteomics, mass spectrometers are used to identify
+compounds and gauge metabolic interactions (reviewed in [Atr05]_).
+
+More and more commonly, studies are employing two or more of these techniques
```
I would  suggest a introduction paragraph early on, tying together the different approaches (nucleic acid sequencing, proteomics, and metabolomics) earlier in this narrative.  It would help set the stage for the details that are contained here.  Also, I think it is worth acknolwedging the pros and cons of each technique.  Though it is true that they can be used to answer different questions, they are also technical and costs advantages to some over others.

adamrp replied:
Although that information would be suitable for a more in-depth review of the available techniques, we feel that such details are not necessary to provide context for the discussion of Qiita.

adina replied:
Agreed, on reread, this sounds good.

adina commented on papers/qiita_development_team/qiita_development_team.rst:
```diff
@@ -0,0 +1,562 @@
+et al. 2014), and the potency of the approach increases as individual studies
+provide more and more detailed characterizations of their samples, enabling
+unanticipated reuse of the data. The power of this approach underscores the
+scientific community’s need for centralized resources for standardized, open
```
Consider deleting "unanticipated".  I do not agree - when I publish data, I anticipate its use and value for others.

adamrp replied:
Done!

adina commented on papers/qiita_development_team/qiita_development_team.rst:
```diff
@@ -0,0 +1,562 @@
+access data.
+
+Here, we present a progress report of Qiita, a multi-omic platform for
+meta-analysis that stresses standardization of data formats, open access to
```
I wish Qiita was presented earlier on....it took about 140 lines for the introduction to the tool and value.

adamrp replied:
We acknowledge that there is a significant amount of background material preceding the mention of Qiita, however we feel that the "top-down" approach we use here provides a clearer motivation for Qiita.

adina commented on papers/qiita_development_team/qiita_development_team.rst:
```diff
@@ -0,0 +1,562 @@
+  for a grant proposal to conduct a more comprehensive analysis. 
+- Type 5: a researcher is interested in performing a tiered study design in
+  which multiple levels of data are generated from the same samples.
+
```
The anecdotes of the types of users is confusing to me.  I'd rather have direct points about what Qiita advantages and when it could be helpful.

adina commented on papers/qiita_development_team/qiita_development_team.rst:
```diff
@@ -0,0 +1,562 @@
+patch version so that what patches need to be applied, if any, can easily be
+determined. Psycopg (http://initd.org/psycopg/) provides Python bindings for
+interacting with PostgreSQL.
+
```
The github contribution.md is nicely done and informative.

adamrp replied:
Thank you for the kind words!

adina commented on papers/qiita_development_team/qiita_development_team.rst:
```diff
@@ -0,0 +1,562 @@
+interactive results visualizations are provided by browser-based tools like
+EMPeror to accelerate the generation and exploration of new hypotheses.
+
+
```
Qiita seems very useful for amplicon type datasets, especially since that is the data that QIIME excels at.  It is not clear to me how transcriptomics, proteomics, and metabolomics data will be integrated?  I signed up for an account, and perhaps this will be clear as I dig into the alpha release but in the text it is not clear right now.

adina replied:
In the context of Scipy review criteria:

Written Quality
Good written quality overall.  I think this could be strengthened by more direct description of Qiita and the gap it currently fills earlier and more clearly.

Technical Content
See comments.  Good description of how python is used.  Ipython notebook integration is neat and I look forward to seeing it.

Novelty and Impact
This could be clarified a bit.  The description of use cases is unclear about how the tool is novel and could lead to direct impact.  Also, the integration of multiomics datasets is not clear.  A flowchart perhaps would be useful.

Verifiability
Code is on github.  I did not look into it deeply to make comments but did sign up for an account to look at alpha release but do not yet have access granted.

Other Requirements
Other requirements seem to be met.

adina commented on papers/qiita_development_team/qiita_development_team.rst:
```diff
@@ -0,0 +1,562 @@
+samples in two different preparation metadata might not overlap (or might
+overlap only partially) since not all samples are analyzed using all available
+techniques.
+
```
A specific example of this might make it clearer to a broader audience on why this is really useful.

adina commented on papers/qiita_development_team/qiita_development_team.rst:
```diff
@@ -0,0 +1,562 @@
+   (http://emperor.microbio.me) serving as the visualization platform for
+   high-dimensional ordination plots, which can be recolored interactively and
+   manipulated using the sample metadata.  
+   
```
Its not clear to me on Qiita how the visualization is integrated.  I see a lot of sample processing but can't figure out how to see analyses.  An example here might be helpful.

adina commented on papers/qiita_development_team/qiita_development_team.rst:
```diff
@@ -0,0 +1,562 @@
+are also available, such as resizing or changing the opacity of arbitrary
+subsets of points. These capabilities shorten the gap between running a purely
+exploratory analysis and producing publication-quality figures.
+
```
Example of EMPeror integration would be fantastic here.

adamrp replied:
Do you think a screen capture of EMPeror running in Qiita would be helpful?

adina replied:
Yes.  In general, I think there is a gap here really showing off Qiita doing the integration.  Hence, the suggestion of some sort of flowchart or specific example.

adamrp replied:
Great! We can certainly include that.

jaberg commented:
Very lay-person review: The paper presents very well-written and clear overview of the qiita tool for analysis of various kinds of data from bio/life-sciences work. It's appropriate for SciPy.

The first suggestion I'd make is in presentation: the paper needs pictures! This tool has a visual interface, and is used to produce a variety of visual summaries of data... I think some pictures would really help tell the story of what (and whom) this tool is for.

I don't feel like I am the target audience for this paper (I'm not scientist working with such data) but if I /were/ I might be interested to know more about how data is routed through the system and how many components are interacting, and generally get an idea of how hard it is to set up this software on my computer and what it will do when I've done it. Our page limit is 8 pages, so there's a little more room to dig into some of the details about how this software works and what it does.  I know that this is a late review, but if you have time to do it, I'd suggest that a little more depth on what Qiita is about would make this a stronger paper.

adamrp commented:
@adina, @jaberg, and @katyhuff:

@ElDeveloper and I believe we have addressed all of the comments left by the reviewers. Thanks for the helpful comments!

ElDeveloper commented:
Thanks @adina, @jaberg and @katyhuff!

katyhuff commented:
Thank YOU! The proceedings will be up very soon. I'll send a link out to
all the authors.

On Thu, Jul 16, 2015 at 3:41 PM, Yoshiki Vázquez Baeza <
notifications@github.com> wrote:

> Thanks @adina <https://github.com/adina>, @jaberg
> <https://github.com/jaberg> and @katyhuff <https://github.com/katyhuff>!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/scipy-conference/scipy_proceedings/pull/145#issuecomment-122123758>
> .
>



-- 
http://katyhuff.github.com
